backend/src/utils/cache.py

```
@@ -20,31 +20,30 @@ def user_cache_key_builder(
     filtered_kwargs = {
-        k: v for k, v in kwargs.items() 
-        if k not in excluded_keys and v is not None
+        k: v for k, v in kwargs.items() if k not in excluded_keys and v is not None
     }
-    
+
     # Create hash of kwargs for cache key uniqueness
     if filtered_kwargs:
         # Convert kwargs to stable string representation
         kwargs_str = str(sorted(filtered_kwargs.items()))
         kwargs_hash = hashlib.md5(kwargs_str.encode()).hexdigest()[:8]
```

Ruff (0.14.10)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)